acl-groovy-dsl/src/main/groovy/javasabr/mqtt/acl/groovy/dsl/builder/UserMatchersFactory.groovy

```diff
@@ -1,27 +1,34 @@
+//file:noinspection unused
 package javasabr.mqtt.acl.groovy.dsl.builder
 
-import javasabr.mqtt.acl.engine.model.matcher.EqualsMatcher
-import javasabr.mqtt.acl.engine.model.matcher.RegexMatcher
-import javasabr.mqtt.acl.engine.model.matcher.StartsWithMatcher
+import javasabr.mqtt.acl.engine.model.matcher.AnyValueMatcher
+import javasabr.mqtt.acl.engine.model.matcher.UserMatchers
 import javasabr.mqtt.acl.engine.model.matcher.ValueMatcher
 
-import java.util.regex.Pattern
-
 abstract class UserMatchersFactory {
 
-  ValueMatcher<String> startsWith(String string) {
-    return new StartsWithMatcher(string);
+  ValueMatcher<String> startsWith(String prefix) {
+    return UserMatchers.startsWith(prefix)
+  }
+
+  ValueMatcher<String> contains(String substring) {
+    return UserMatchers.contains(substring)
   }
 
   ValueMatcher<String> eq(String string) {
-    return new EqualsMatcher(string);
+    return UserMatchers.eq(string)
   }
 
   ValueMatcher<String> regex(String string) {
-    return new RegexMatcher(Pattern.compile(string));
+    return UserMatchers.regex(string)
+  }
+
+  ValueMatcher<String> anyValue() {
+    return ValueMatcher.MATCH_ANY_STRING
   }
 
-  ValueMatcher<?> anyValue() {
-    return ValueMatcher.MATCH_ANY;
+  ValueMatcher<String> replaceMatcherIfNeed(ValueMatcher<String> it) {
+    return it instanceof AnyValueMatcher ? ValueMatcher.MATCH_ANY_STRING : it
```

Comment:
It seems to be not required here as there is already expected instance

https://github.com/JavaSaBr/mqtt-broker/blob/430922f2a145da71890b6522a5ddc5f3940ddd21/acl-groovy-dsl/src/main/groovy/javasabr/mqtt/acl/groovy/dsl/builder/UserMatchersBuilder.groovy#L46-L50